client/landing/stepper/declarative-flow/internals/steps-repository/sensei-setup/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { Gridicon } from '@automattic/components';
+import { SenseiStepContainer } from '@automattic/onboarding';
+import { useDesktopBreakpoint } from '@automattic/viewport-react';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useCallback, useState } from '@wordpress/element';
+import { useI18n } from '@wordpress/react-i18n';
+import classnames from 'classnames';
+import { CSSProperties } from 'react';
+import FormRadioWithThumbnail from 'calypso/components/forms/form-radio-with-thumbnail';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { ONBOARD_STORE } from '../../../../stores';
+import { Title, Label, Input, Button } from './components';
+import type { StyleVariation } from 'calypso/../packages/design-picker/src/types';
+import type { Step } from 'calypso/landing/stepper/declarative-flow/internals/types';
+import './style.scss';
+
+type ThemeStyle = {
+	name: string;
+	title: string;
+};
+
+const styles: ThemeStyle[] = [
+	{ name: 'green', title: 'Green' },
+	{ name: 'blue', title: 'Blue' },
+	{ name: 'dark', title: 'Dark' },
+	{ name: 'gold', title: 'Gold' },
+];
+
+const ThemeStylePreviews = ( {
+	styles,
+	active,
+}: {
+	styles: ThemeStyle[];
+	active: string;
+	siteTitle?: string;
+} ) => {
+	const index = 1 + styles.findIndex( ( style ) => active === style.name );
+	return (
+		<div className="sensei-theme-previews">
+			<div className="sensei-theme-list" style={ { '--index': index } as CSSProperties }>
+				{ styles.map( ( { name } ) => {
+					return (
+						<div
+							key={ name }
+							className={ classnames( 'sensei-theme-item', { selected: name === active } ) }
+						>
+							<div className={ `sensei-theme ${ name }` }></div>
+						</div>
+					);
+				} ) }
+			</div>
+		</div>
+	);
+};
+
+const SenseiSetup: Step = ( { navigation } ) => {
+	const { __ } = useI18n();
+	const isDesktop = useDesktopBreakpoint();
+
+	const initialSiteTitle = useSelect( ( select ) =>
+		select( ONBOARD_STORE ).getSelectedSiteTitle()
+	);
+	const [ siteTitle, setSiteTitle ] = useState< string >( initialSiteTitle );
+	const [ checked, setChecked ] = useState< string >( 'green' );
+
+	const handleChange = ( event: React.ChangeEvent< HTMLInputElement > ) => {
+		setChecked?.( event.target.value );
+	};
+
+	const { goNext } = navigation;
+	const dispatch = useDispatch( ONBOARD_STORE );
+	const handleSubmit = useCallback( () => {
+		dispatch.setSiteTitle( siteTitle );
+		const variation = styles.find( ( style ) => style.name === checked ) || styles[ 0 ];
+		dispatch.setSelectedStyleVariation( {
+			slug: variation.name,
+			title: variation.title,
+		} as StyleVariation );
+		goNext?.();
```

Comment:
You can call submit instead. `goNext` is deprecated.

Reply:
Fixed in f2da75ffa9b20b20abf3b1f4b8c14da042610ddb